plugins/auto-sizes/includes/improve-calculate-sizes.php

```diff
@@ -213,6 +221,10 @@ function auto_sizes_calculate_better_sizes( int $id, $size, string $align, int $
 			 * on the server. Otherwise, we need to rely on CSS functions.
 			 */
 			if ( str_ends_with( $layout_width, 'px' ) ) {
+				// First remove 'px' from width.
+				$layout_width = str_replace( 'px', '', $layout_width );
+				// Convert to float for better precision.
+				$layout_width = (float) $layout_width * $container_relative_width;
 				$layout_width = sprintf( '%dpx', min( (int) $layout_width, $image_width ) );
 			} else {
 				$layout_width = sprintf( 'min(%1$s, %2$spx)', $layout_width, $image_width );
```

Comment:
Is it allowed to have relative values like `%` in the `sizes` attribute? Or what is the rationale for having this `min()` value here?

Reply:
See #1737 for context why it was added.

Reply:
Makes sense. From that perspective though, wouldn't we be able to also use non-`px` values for the multiplication above? Right now the `container_relative_width` is only used if the unit is `px`.

For which units can this `min()` function be used in the `sizes` attribute? I would expect it to be for `em`, `rem` etc - but what about `%`? Since `%` needs to know the container width, I wonder whether that's even supported or not.

@joemcgill Maybe you have additional context?

Reply:
So basically `px`, `em`, `rem`, and `vw` are valid if I understand correctly? If so, ideally we should support all of these in our implementation and the calculation.

Adding support for the units other than `px` in a follow up PR works for me. It would be good though to add a `TODO` comment here above the relevant lines to clarify support for these other units is still to be added.

Reply:
TODO comment added in https://github.com/WordPress/performance/pull/1818/commits/80cc029c91d931eb90340f3ab1ec8c7b8a6186a0